### ISSUE TITLE
error using psite_info output

### ISSUE THREAD

**Issue by cjdecker** [LabTranslationalArchitectomics/riboWaltz#18]
Hi, I am using ribowaltz on two bam files generated by aligning reads to human protein coding transcripts obtained from gencode. After running the psite_info function I am able to run downstream functions on one of the samples but for the second sample I keep getting the following error:
Error in as.character(transcript) %in% c_transcripts : 
  object 'transcript' not found

I obtained both the fasta file (gencode.v29.pc_transcripts.fa) and gtf file (gencode.v29.annotation.gtf) from gencode. I first tried to use ribowaltz using alignments to the original gencode.v29.pc_transcripts.fa file but got error because the transcript name in the alignment included the entire description line in the fasta file. So, I made a modified the fasta file so that only the transcript ID was in the description line and used that modified fasta file for the alignment.

In case it helps resolve the error issue above I get the following warning when I run the create_annotation function:
Warning message:
In .get_cds_IDX(type, phase) :
  The "phase" metadata column contains non-NA values for features of type stop_codon. This information was ignored.

I have two questions. 1. can you help me resolve the issue with the psite_info function. 2. Instead of using gencode as a source, what source would you recommend for obtaining human annotation/genome data and what method would you recommend to generate a transcript fasta file? 

Thanks!
Carolyn

**Comment by fabiolauria** [comment]
Hi Carolyn,
thank for using riboWaltz and reporting these issues.
Let me address your problems point by point:

> [...] for the second sample I keep getting the following error:
> Error in as.character(transcript) %in% c_transcripts :
> object 'transcript' not found

It's odd you get this error only for one of your samples. It suggests it's not a problem of riboWaltz but of the input data. Did you check if your list of samples actually includes both datasets, that both are data.tables and that the second one is a non-empty object? The error above usually arises if there are no "transcript" to match c_transcript i.e. there are no lines in the data.table. Which is the function throwing it? 

> [...]but got error because the transcript name in the alignment included the entire description line in the fasta file. So, I made a modified the fasta file so that only the transcript ID was in the description line and used that modified fasta file for the alignment.

Thank you for pinpointing this issue about the headers of FASTA files. I'll add to _bamtolist_ an option for removing everything after a specified symbol, so that there will be no need of manually modify the file. This should solve the problem even for GENCODE-derived FASTAs.

> In case it helps resolve the error issue above I get the following warning when I run the create_annotation function:
> Warning message:
> In .get_cds_IDX(type, phase) :
> The "phase" metadata column contains non-NA values for features of type stop_codon. This information was ignored.

Don't worry about this, it is not actually a problem. I know most of the time the warning comes out because feature _stop_codon_ is rarely included in GTF files. But it is not required for _create_annotation_'s purposes.

> 1. can you help me resolve the issue with the psite_info function

I need more information for this (see my first comment).

> 2. Instead of using gencode as a source, what source would you recommend for obtaining human annotation/genome data and what method would you recommend to generate a transcript fasta file?

I think GENCODE is a good place to download transcriptomes and genomes sequences coupled with their correct GTFs. It's clear, easy and always updated. But if you need FASTAs for organisms different from human and mouse, I would go for the Ensembl's [FTP site](ftp://ftp.ensembl.org/pub/).

I hope this helps.

Bests
Fabio

**Comment by cjdecker** [comment]
HI Fabio,

Thanks for your prompt reply. I agree that the issue is something about one of my datasets.
In fact, to prepare information to send to you to help resolve the issue I had copied both bam files to a new directory and now ribowaltz is working fine on both datasets. Perhaps the original bam file was corrupted in someway…

Thank you for developing ribowaltz. This is my first time analyzing riboprofiling data and ribowaltz has been the best documented and understandable package I have tried to use to examine my test datasets.

Carolyn
